clientapi/routing/register.go

```diff
@@ -759,13 +760,16 @@ func completeRegistration(
 			JSON: jsonerror.BadJSON("missing password"),
 		}
 	}
-
+	accType := userapi.AccountTypeUser
+	if appserviceID != "" {
+		accType = userapi.AccountTypeAppService
```

Comment:
If you look [here](https://github.com/matrix-org/dendrite/blob/a84f50f4fb0963e15b6aca60f658bed45b779127/clientapi/routing/register.go#L966) you can see we call `completeRegistration` during shared secret auth but ignore the [admin flag[(https://github.com/matrix-org/dendrite/blob/a84f50f4fb0963e15b6aca60f658bed45b779127/clientapi/routing/register_secret.go#L26). We should really move the `accType` to be an arg to `completeRegistration`.